src/app/components/quiz.tsx

```diff
@@ -0,0 +1,110 @@
+"use client";
+
+import { useState } from "react";
+import clsx from "clsx";
+
+export interface QuizParams {
+  className?: string;
+  question: string;
+  hint?: string;
+  answers: string[];
+  correctAnswer: number;
+  hintAnswers?: string[];
+  onSelect?: (selection: number, isDone: boolean) => void;
+  // If enabled, the user only has a single try. After selection, it will show the solution to the user.
+  showCorrectAnswer?: boolean;
+}
+
+/**
+ * A quiz component.
+ * @param quiz The quiz parameters
+ */
+export default function Quiz(quiz: QuizParams) {
+  const [selection, setSelection] = useState<number>();
+
+  function onClick(index: number) {
+    if (quiz.showCorrectAnswer && selection != undefined) return;
+    setSelection(index);
+    const isDone = quiz.showCorrectAnswer
+      ? index != undefined
+      : index == quiz.correctAnswer;
+    console.log("Quiz", isDone, quiz.showCorrectAnswer, selection);
+    quiz.onSelect?.(index, isDone);
+  }
+
+  const selectionClass = (index: number) => {
+    if (
+      quiz.showCorrectAnswer &&
+      selection != undefined &&
+      index == quiz.correctAnswer
+    ) {
+      return "bg-lime-500";
+    } else {
+      return selection == index
+        ? index == quiz.correctAnswer
+          ? "bg-lime-500"
+          : "bg-red-500"
+        : "";
+    }
+  };
+
+  const showCursorClass = () =>
+    quiz.showCorrectAnswer && selection != undefined ? "cursor-auto" : null;
+
+  return (
+    <div className={quiz.className}>
+      <h3 className="text-xl font-semibold mb-2">{quiz.question}</h3>
+      <p className={"mb-10"}>{quiz.hint}</p>
+      <div className={"answers grid grid-cols-2 gap-4"}>
+        {quiz.answers.map((answer, index) => (
+          <button
+            key={index}
+            className={clsx(
+              "button transition-colors ",
+              selectionClass(index),
+              showCursorClass(),
+            )}
+            onClick={() => {
+              onClick(index);
+            }}
+          >
+            {answer}
+          </button>
```

Comment:
can you extract the button in a seperate component? then you can specify the style of the button using tailwindcss.

Reply:
Can't we just create a default style for the button? If we would create a component, we would have to publish all the events from the dom button element we might need manually. I think this would unnececssarily overengineer the project 😅

Reply:
https://tailwindcss.com/docs/reusing-styles#extracting-classes-with-apply

Reply:
I think the button component would be a great idea, because we could reuse the button in the project.  To your message: We design the button component as generic as possible

Reply:
If you insist on a full blown Button React component, we can do it, but I feel we overcomplicate stuff that we don't have to 😅

Reply:
I thought about it, and I think the abstraction wouldn't give us any benifits. In terms of code style, the code would be just the same, since the only abstraction would be the tailwind classes. Moreover we would need to maintain the component, if we need more functionality. I think in this case it would be easier and cleaner to create a separate component, that implements those required features.